lib/aur-build

```diff
@@ -59,93 +56,95 @@ var_tmp=$(mktemp -d "${TMPDIR:-/var/tmp}/$argv0".XXXXXXXX)
 trap 'trap_exit' EXIT
 trap 'exit' INT
 
-opt_short='a:d:r:C:D:M:cfNRsv'
-opt_long=()
+## option parsing
+opt_short='a:d:p:r:D:cfNRsv'
+opt_long=('arg-file:' 'chroot' 'database:' 'force' 'path:' 'root:'
+          'sign' 'verify' 'directory:' 'no-sync' 'pacman-conf:'
+          'makepkg-conf:' 'remove')
+
 if ! parseopts "$opt_short" "${opt_long[@]}" -- "$@"; then
     usage
 fi
 set -- "${OPTRET[@]}"
 
-unset db_name db_root queue
+unset queue db_name db_path db_root
 while true; do
     case "$1" in
-        -a) shift; queue=$1 ;;
-        -c) chroot=1 ;;
-        -f) overwrite=1 ;;
-        -N) no_sync=1 ;;
-        -r) shift; db_root=$1 ;;
-        -d) shift; db_name=$1
-            chroot_args+=(-d "$1") ;;
-        -C) shift; chroot_args+=(-C "$1") ;;
-        -D) shift; chroot_args+=(-D "$1") ;;
-        -M) shift; chroot_args+=(-M "$1") ;;
-        -R) repo_add_args+=(-R) ;;
-        -s) sign_pkg=1
-            repo_add_args+=(-s) ;;
-        -v) repo_add_args+=(-v) ;;
+        -a|--arg-file)  shift; queue=$1 ;;
+        -p|--path)      shift; db_path=$1 ;;
+        -d|--database)  shift; db_name=$1
+                        repo_args+=(-d "$1") ;;
+        -r|--root)      shift;
+                        repo_args+=(-r "$1") ;;
+        -c|--chroot)    chroot=1 ;;
+        -f|--force)     overwrite=1 ;;
+        -s|--sign)      sign_pkg=1
+                        repo_add_args+=(-s) ;;
+        -v|--verify)    repo_add_args+=(-v) ;;
+        -D|--directory) shift; chroot_args+=(-D "$1") ;;
+        -N|--no-sync)   no_sync=1 ;;
+        -R|--remove)    repo_add_args+=(-R) ;;
+        --pacman-conf)  shift; chroot_args+=(-C "$1") ;;
+        --makepkg-conf) shift; chroot_args+=(-M "$1") ;;
         --) shift; break ;;
     esac
     shift
 done
 unset opt_short opt_long OPTRET
 
-if [[ -n ${db_name=$AUR_REPO} ]]; then
-    if [[ -z ${db_root=$AUR_DBROOT} ]]; then
-        db_root=$(pacconf --single --repo="$db_name" Server)
-        db_root=${db_root#*//}
+# reset default makechrootpkg arguments
+if (($#)); then
+    if ((chroot)); then
+        makechrootpkg_args=("$@")
+    else
+        makepkg_args=("$@")
     fi
+fi
 
-     # .db symbolic link to archive
-    db_path=$(realpath -e -- "$db_root/$db_name".db)
-    db_root=$(realpath -e -- "$db_root")
-else
-    error "$argv0: database: missing argument"
-    usage
+if type -P xdelta3 >/dev/null; then
+    repo_add_args+=(-d)
+fi
+
+if ! [[ ${db_path=$AUR_DBPATH} ]]; then
```

Comment:
I am a bit confused here, aur-repo only uses `$AUR_REPO` and `$AUR_DBROOT` (I guess because you want to use the name of the repo before symlink resolution), while aur-build checks `$AUR_DBPATH` too. But then if the user set both `$AUR_DBPATH` and `$AUR_REPO` there could be an inconsistency if AUR_REPO does not refer to the correct database.

This is unlikely to happen since `aur sync` will call `aur build` with the `-p` option anyway, but i find strange that only `aur-build` looks for a `$AUR_DBPATH` env var.

Reply:
I had a similar concern. The idea was that `aur-build` should remain usable if `aur-repo` breaks for some reason, though having a possible inconsistency is arguably worse.

Reply:
Okay, my concern was that the symlink to `foo.db` was resolved twice (once in `aur-sync` through `aur-repo --status-file`, and once in `aur-build`). However, there's no real reason for `aur-repo` to already resolve this symbolic link (whose target may change at any point).